packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -39,28 +40,44 @@ export interface MigrationFile {
 
 type MIGRATION_STEP = typeof MIGRATION_STEPS[number]
 
+const flags = {
+  hasCustomIntegrationFolder: false,
+  hasCustomIntegrationTestFiles: false,
+
+  hasCustomComponentFolder: false,
+  hasCustomComponentTestFiles: false,
+
+  hasCustomSupportFile: false,
+  hasComponentTesting: true,
+  hasE2ESpec: true,
+  hasPluginsFile: true,
+} as const
```

Comment:
Is the additional cast here needed?
```suggestion
}
```

Reply:
`as const` is needed here to help typescript making a type out of the keys.

see this blog post:
https://www.bscotch.net/post/typescript-as-const